src/components/NodeUI.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Box, Input, Stack, HStack, VStack, Editable, EditableInput, EditableTextarea, EditablePreview, IconButton } from '@chakra-ui/react';
+import { AddIcon, CloseIcon } from '@chakra-ui/icons';
+import { Node, addListType, deleteListType } from '../types/node';
+
+export function NodeUI({ item, addList, deleteList }:
+  {item: Node, addList: addListType, deleteList: deleteListType}) {
```

Comment:
propsの部分は
```suggestion
  NodeUI({ item, addList, deleteList }: Props
```
のようにするのが良いかと思います。

functionを定義する前に
```
type Props = {
  item: Node;
  addList: AddListType;
  deleteListType: DeleteListType;
}
```
のようにしておくとfunction内はすっきりします。